core/trino-main/src/main/java/io/trino/sql/planner/optimizations/PropertyDerivations.java

```diff
@@ -562,6 +564,14 @@ public ActualProperties visitExchange(ExchangeNode node, List<ActualProperties>
             // are computed for local exchanges.
             // TODO: implement full properties for local exchanges
             if (node.getScope() == LOCAL) {
+                if (inputProperties.size() == 1) {
+                    ActualProperties inputProperty = inputProperties.get(0);
+                    if (inputProperty.isEffectivelySingleStream() && node.getOrderingScheme().isEmpty()) {
+                        // Retain source local properties
+                        localProperties.addAll(inputProperty.getLocalProperties());
+                    }
+                }
+
```

Comment:
This change seems like an optimization unrelated to this commit. If so, move it to a separate commit

Reply:
This is extracted as PR https://github.com/trinodb/trino/pull/7589
Please take a look at that one as well.
It's necessary for this feature, otherwise we loose ordering property if there is an exchange between the node producing sorted output and partial topN.
E.g.: TestPartialTopNWithPresortedInput#testWithSortedWindowFunction

Reply:
That shouldn't matter, as the "is order sensitive" flag in the LimitNode is only used to make decisions about whether the node can be pushed through another node (i.e., can only be done if the source isn't the one that establishes an ordering)

Reply:
Without this change, ValidateLimitWithPresortedInput starts failing on `TestPartialTopNWithPresortedInput#testWithSortedWindowFunction`. 
First the TopN -> Order sensitive Limit conversion takes place due to sorted input from WindowNode, then a local exchange gets introduced between them. Now the validation in ValidateLimitWithPresortedInput finds that StreamProperties of order sensitive Limit's source is no longer satisfying the desired input ordering.